src/Configuration/Configuration.php

```diff
@@ -36,12 +40,64 @@ public function bind(Configuration $parent): void
 
     /**
      * @param mixed $value
+     * @param callable $callback
      */
-    public function set(string $name, $value): void
+    public function set(string $name, $value, callable $callback = null): void
     {
+        if (is_callable($callback)) {
+            $this->validations[$name] = $callback;
+        }
+        $callingFile = pathinfo(debug_backtrace()[1]['file'], PATHINFO_FILENAME);
+        if ($callingFile === 'deploy') {
+            if ($this->has($name)) {
+                $this->verifyConfigurationValue($name, $value);
+            } else {
+                $this->verifyConfigurationName($name);
+            }
+        }
         $this->values[$name] = $value;
     }
 
+    /**
+     * @param string $name
+     * @param mix $value
+     */
+    protected function verifyConfigurationValue(string $name, $value): void
+    {
+        if ($this->parent) {
+            $definedCallable = &$this->parent;
+        } else {
+            $definedCallable = &$this;
+        }
+        if (array_key_exists($name, $definedCallable->validations) 
```

Comment:
If the callable is a native function, i.e. `is_array` it would be nice to have a check if the function exists and generate an exception telling this to the user.

Reply:
Oh, the `set` function argument is defined as `callable`, so if the function doesn't exist, it will already throw an error. :)

Reply:
That's right. Didn't get that!